profiler/src/ProfilerEngine/Datadog.Profiler.Native/ClrEventsParser.cpp

```diff
@@ -100,6 +104,116 @@ void ClrEventsParser::ParseEvent(
+    //     <data name="SampledByteOffset" inType="win:UInt64" />
+
+    // DumpBuffer(pEventData, cbEventData);
+
+    ULONG offset = 0;
+    if (!EventsParserHelper::Read(payload.AllocationAmount, pEventData, cbEventData, offset))
+    {
+        return false;
+    }
+    if (!EventsParserHelper::Read(payload.AllocationKind, pEventData, cbEventData, offset))
+    {
+        return false;
+    }
+    if (!EventsParserHelper::Read(payload.ClrInstanceId, pEventData, cbEventData, offset))
+    {
+        return false;
+    }
+    if (!EventsParserHelper::Read(payload.AllocationAmount64, pEventData, cbEventData, offset))
+    {
+        return false;
+    }
+    if (!EventsParserHelper::Read(payload.TypeId, pEventData, cbEventData, offset))
+    {
+        return false;
+    }
+    payload.TypeName = EventsParserHelper::ReadWideString(pEventData, cbEventData, &offset);
+    if (payload.TypeName == nullptr)
+    {
+        return false;
+    }
+    if (!EventsParserHelper::Read(payload.HeapIndex, pEventData, cbEventData, offset))
+    {
+        return false;
+    }
+    if (!EventsParserHelper::Read(payload.Address, pEventData, cbEventData, offset))
+    {
+        return false;
+    }
+    if (!EventsParserHelper::Read(payload.ObjectSize, pEventData, cbEventData, offset))
+    {
+        return false;
+    }
+
+    return true;
+}
+
+bool ClrEventsParser::ParseAllocationSampledEvent(ULONG cbEventData, LPCBYTE pEventData, AllocationSampledPayload& payload)
+{
+    // <template tid = "AllocationSampled">
+    //    <data name = "AllocationKind" inType = "win:UInt32" / >
+    //    <data name = "ClrInstanceID" inType = "win:UInt16" / >
+    //    <data name = "TypeID" inType = "win:Pointer" / >
+    //    <data name = "TypeName" inType = "win:UnicodeString" / >
+    //    <data name = "Address" inType = "win:Pointer" / >
+    //    <data name = "ObjectSize" inType = "win:UInt64" / >
+    //    <data name = "SampledByteOffset" inType = "win:UInt64" / >
+    //
+
+    // DumpBuffer(pEventData, cbEventData);
+
+    ULONG offset = 0;
+    if (!EventsParserHelper::Read(payload.AllocationKind, pEventData, cbEventData, offset))
+    {
+        return false;
+    }
+    if (!EventsParserHelper::Read(payload.ClrInstanceId, pEventData, cbEventData, offset))
+    {
+        return false;
+    }
+    if (!EventsParserHelper::Read(payload.TypeId, pEventData, cbEventData, offset))
+    {
+        return false;
+    }
+    payload.TypeName = EventsParserHelper::ReadWideString(pEventData, cbEventData, &offset);
+    if (payload.TypeName == nullptr)
+    {
+        return false;
+    }
+    if (!EventsParserHelper::Read(payload.Address, pEventData, cbEventData, offset))
+    {
+        return false;
+    }
+    if (!EventsParserHelper::Read(payload.ObjectSize, pEventData, cbEventData, offset))
+    {
+        return false;
+    }
+    if (!EventsParserHelper::Read(payload.SampledByteOffset, pEventData, cbEventData, offset))
```

Comment:
Wouldn't it be simpler to group all in a single if statement?

like this for exemple:
```suggestion
    if (!EventsParserHelper::Read(payload.Address, pEventData, cbEventData, offset) ||
    !EventsParserHelper::Read(payload.ObjectSize, pEventData, cbEventData, offset) ||
  !EventsParserHelper::Read(payload.SampledByteOffset, pEventData, cbEventData, offset))
```

there might be a good reason against it, but I don't see it 🙂

Reply:
Makes a lot of sense thanks!